### DISCUSSION TITLE
Visualizing Network Path Activity - Showcase

### DISCUSSION THREAD

**Discussion by avarga** [inet-framework/inet#593] [Showcases]
Discuss the "Visualizing Network Path Activity" visualization showcase here. Ideas and recommendations are welcome.

**Comment by ManiAm** [comment]
Question 1:

>> NetworkRouteVisualizer currently observes packets that pass through the network layer (i.e. carry data from/to higher layers), but not those that are internal to the operation of the network layer protocol. That is, packets such as ARP, although potentially useful, will not trigger the visualization. **_Visualizing such packets may be implemented in future INET revisions._**

ARP messages are encapsulated into Ethernet frames and thus have nothing to do with IP packets. They are technically data-link frames, so why not displaying them with 'data link activity visualizer'?

Edit: In fact in 'Visualizing Data Link Activity', arp messages are also visualized. So what do you mean by 'Visualizing such packets may be implemented in future INET revisions.'

**Comment by ManiAm** [comment]
Question 2: In the 'Visualizing Network Path Activity in a Mobile Ad-Hoc Network' scenario, how come the AODV-req frames from the source is received by all nodes in the network? The radio coverage of the source (shown in blue) does not encompass all nodes!

**Comment by levy** [comment]
I think the reference to being implemented in the future is wrong and should be fixed.

Receiving a message according to the lowest level (aka. physical transmission medium) does not imply the packet will be successfully received by the receiver. Nevertheless, receivers are interested in signals even if they cannot be received correctly, because some MAC protocols act on them.

**Comment by ManiAm** [comment]
>> Receiving a message according to the lowest level (aka. physical transmission medium) does not imply the packet will be successfully received by the receiver. Nevertheless, receivers are interested in signals even if they cannot be received correctly, because some MAC protocols act on them.

So what is the blue circle around the nodes? I thought it is the maximum interference distance that is calculated based on the free space loss and it would be the maximum theoretical distance that a frame can reach (healthy or with error due to collision, thermal noise, etc.). Any point beyond the 'max interference distance', or blue circle, is not decodable.

Edit: I remember in MiXiM (R.I.P), the maximum interference range is calculated first in the sending node, and then all nearby nodes that are inside this circle are selected and the frame is sent directly to them (using sendDirect). Each receiver node decides independently whether the frame is healthy or not. I have not checked how the PHY layer of INET is implemented, but the current visualization approach seems confusing to me, and apparently to others too:

https://github.com/inet-framework/inet-tutorials/issues/1

**Comment by levy** [comment]
The blue circle is the estimation of the maximum communication range solely based on free space path loss. The circle is for a particular transmitter assuming the highest transmission power it's capable of and the most sensitive receiver in the network is receiving the signal. Receivers outside the communication range cannot successfully receive packets (again: approximation). Nevertheless, receivers may still be interested in packets outside the communication range, because they may cause certain state changes. For example, if the receive signal power is just below the threshold of successful reception, the signal may still be needed to determine if the channel is clear to send. Moreover, the radioMedium can be instructed to reduce the number of packets to receivers if it's known to not cause any effect on them, see rangeFilter parameter. To summarize, the red line drawn between two wireless nodes doesn't mean successful packet reception. Only a link visualizers display accurately successful link level activity. I hope this helps.

**Comment by windy32** [comment]
configuration.xml and changeRoute.xml and not provided at the bottom of the documentation.

As a result the 4th model cannot be started.
